lms/djangoapps/commerce/tests/test_utils.py

```diff
@@ -105,3 +114,119 @@ def test_get_checkout_page_url(self, skus):
+            settings.ECOMMERCE_API_URL + 'refunds/',
+            status=201,
+            body='[1]',
+            content_type='application/json'
+        )
+        httpretty.register_uri(
+            httpretty.PUT,
+            settings.ECOMMERCE_API_URL + 'refunds/1/process/',
+            status=200,
+            body=json.dumps({
+              "id": 9,
+              "created": "2017-12-21T18:23:49.468298Z",
+              "modified": "2017-12-21T18:24:02.741426Z",
+              "total_credit_excl_tax": "100.00",
+              "currency": "USD",
+              "status": "Complete",
+              "order": 15,
+              "user": 5
+            }),
+            content_type='application/json'
+        )
+        course_entitlement = CourseEntitlementFactory()
+        refund_success = refund_entitlement(course_entitlement)
+        assert refund_success
+
+    @httpretty.activate
+    @patch('lms.djangoapps.commerce.utils._send_refund_notification', return_value=True)
+    def test_ecommerce_refund_failed_process_notification_sent(self, mock_send_notification):
+        httpretty.register_uri(
+            httpretty.POST,
+            settings.ECOMMERCE_API_URL + 'refunds/',
+            status=201,
+            body='[1]',
+            content_type='application/json'
+        )
+        httpretty.register_uri(
+            httpretty.PUT,
+            settings.ECOMMERCE_API_URL + 'refunds/1/process/',
+            status=400,
+            body='{}',
+            content_type='application/json'
+        )
+        course_entitlement = CourseEntitlementFactory.create(mode=CourseMode.VERIFIED)
+        refund_success = refund_entitlement(course_entitlement)
+        assert mock_send_notification.is_called
+        call_args = list(mock_send_notification.call_args)
+        assert call_args[0] == (course_entitlement, [1])
+        assert refund_success
+
+    @httpretty.activate
+    @patch('lms.djangoapps.commerce.utils._send_refund_notification', return_value=True)
+    def test_ecommerce_refund_send_notification_failed(self, mock_send_notification):
+        httpretty.register_uri(
+            httpretty.POST,
+            settings.ECOMMERCE_API_URL + 'refunds/',
+            status=201,
+            body='[1]',
+            content_type='application/json'
+        )
+        httpretty.register_uri(
+            httpretty.PUT,
+            settings.ECOMMERCE_API_URL + 'refunds/1/process/',
+            status=400,
+            body='{}',
+            content_type='application/json'
+        )
+        mock_send_notification.side_effect = NotImplementedError
+        course_entitlement = CourseEntitlementFactory.create(mode=CourseMode.VERIFIED)
+        refund_success = refund_entitlement(course_entitlement)
+
+        assert mock_send_notification.is_called
+        call_args = list(mock_send_notification.call_args)
+        assert call_args[0] == (course_entitlement, [1])
+        assert not refund_success
```

Comment:
It might worthwhile to consider adding a test for the `mode != 'verified'` case.
https://github.com/edx/edx-platform/pull/16934/files#diff-4394a757d8d82dd3dc4351b59a073065R261